frontend/src/metabase/admin/settings/components/EmbeddingSettings/EmbeddingSettingsSwitch/EmbeddingSettingsSwitch.tsx

```diff
@@ -0,0 +1,29 @@
+import { t } from "ttag";
+
+import { Switch, Text } from "metabase/ui";
+import type { SettingKey } from "metabase-types/api";
+
+import { useEmbeddingSetting } from "../hooks";
+
+export const EmbeddingSettingsSwitch = ({
+  settingKey,
+  switchLabel,
+}: {
+  settingKey: SettingKey;
+  switchLabel: string;
+}) => {
+  const [setting, handleChange] = useEmbeddingSetting({ key: settingKey });
+  const isSettingEnabled = Boolean(setting.value);
+
+  return setting.is_env_setting ? (
+    <Text color="var(--mb-color-text-secondary)">{t`Set via environment variable`}</Text>
```

Comment:
I wonder if we can use `SwitchWithSetByEnvVar` for this instead, since it looks quite similar.

Reply:
I agree.

Reply:
agreed as well. I'll see what I can do

Reply:
that has been solved by completely getting rid of the other switch